functable.c

```diff
@@ -393,6 +393,12 @@ Z_INTERNAL uint32_t crc32_stub(uint32_t crc, const unsigned char *buf, uint64_t
 #elif defined(S390_CRC32_VX)
     if (s390_cpu_has_vx)
         functable.crc32 = &s390_crc32_vx;
+#elif defined(X86_PCLMULQDQ_CRC)
+    /* For lens < 64, crc32_byfour method is faster. The CRC32 instruction for
+     * these short lengths might also prove to be effective */
+    if (x86_cpu_has_pclmulqdq && len >= 64) {
+        functable.crc32 = &crc32_pclmulqdq;
```

Comment:
Checking `len` here doesn't work as expected as the stub is called only once. The check should be inside the callback function and the callback function should call the fallback function if `len` there is too small...

Reply:
Yep, fixed it.